models/Asset.php

```diff
@@ -966,77 +966,107 @@ public function delete(bool $isNested = false): void
             throw new Exception('root-node cannot be deleted');
         }
 
-        $this->dispatchEvent(new AssetEvent($this), AssetEvents::PRE_DELETE);
-
-        $this->beginTransaction();
+        if (!$isNested) {
+            $deletedPaths = [];
+            $deletedIds = [];
+        }
 
         try {
-            $this->closeStream();
+            $this->dispatchEvent(new AssetEvent($this), AssetEvents::PRE_DELETE);
 
-            // remove children
-            if ($this->hasChildren()) {
-                foreach ($this->getChildren() as $child) {
-                    $child->delete(true);
-                }
-            }
+            $maxRetries = 5;
+            for ($retries = 0; $retries < $maxRetries; $retries++) {
+                $this->beginTransaction();
 
-            // Dispatch Symfony Message Bus to delete versions
-            Pimcore::getContainer()->get('messenger.bus.pimcore-core')->dispatch(
-                new VersionDeleteMessage(Service::getElementType($this), $this->getId())
-            );
+                try {
+                    $this->closeStream();
+
+                    // remove children
+                    if ($this->hasChildren()) {
+                        foreach ($this->getChildren() as $child) {
+                            $child->delete(true);
+                            $deletedPaths[] = $child->getRealPath() . $child->getId();
+                            $deletedIds[] = $child->getId();
+                        }
+                    }
```

Comment:
there is something different, in the Assets it seems able to include children in the deletion, but it's done per element
https://github.com/pimcore/admin-ui-classic-bundle/blob/396d3f6d1a58d90a43269b0bc57968f751d18b5b/src/Controller/Admin/Asset/AssetController.php#L659-L671